pennylane/gradients/parameter_shift.py

```diff
@@ -1463,13 +1464,8 @@ def _param_shift_new(
     gradient_tapes.extend(g_tapes)
 
     if unsupported_params:
-        # If there are unsupported parameters, we must process
-        # the quantum results separately, once for the fallback
-        # function and once for the parameter-shift rule, and recombine.
-        def processing_fn(results):
-            unsupported_grads = fallback_proc_fn(results[:fallback_len])
-            supported_grads = fn(results[fallback_len:])
 
+        def _single_grad(unsupported_grads, supported_grads):

```

Comment:
Maybe also worth renaming this function, because that could be single measurement, param or shots batch

Reply:
Clarified.